src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticService.cs

```diff
@@ -0,0 +1,118 @@
+
+            var rootOne = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var editor = new SyntaxEditor(rootOne, s_generator);
+
+
+            foreach (var referenceSymbol in arrayNode)
+            {
+                foreach (var location in referenceSymbol.Locations)
+                {
+                    var root = await location.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                    var syntaxNode = root.FindNode(location.Location.SourceSpan); //Node for the identifier syntax
+
+
+                    var invocation = (syntaxNode as IdentifierNameSyntax).Parent as InvocationExpressionSyntax;
+                    if (invocation == null)
+                    {
+                        return document;
+                    }
+
+                    var argList = invocation.ArgumentList;
+                    List<ArgumentSyntax> x = new List<ArgumentSyntax>();
+
+                    foreach (var parameter in parameters)
+                    {
+
+                        var newArgument = GenerateArgument(parameter, parameter.Name, false);
+                        x.Add(newArgument as ArgumentSyntax);
+                    }
+
+
+                    var newArgList = argList.WithArguments(argList.Arguments.AddRange(x));
+                    var newInvocation = invocation.WithArgumentList(newArgList);
+
+                    editor.ReplaceNode(invocation, newInvocation);
+
+                }
+            }
+            //Updates the declaration with the variables passed in
+            var newLF = CodeGenerator.AddParameterDeclarations(localfunction, parameters, workspace);
+
+            //Adds the modifier static
+            var modifiers = DeclarationModifiers.From(localFunctionSymbol).WithIsStatic(true);
+            var LFWithStatic = s_generator.WithModifiers(newLF, modifiers);
+
+            editor.ReplaceNode(localfunction, LFWithStatic);
+
+            var newRoot = editor.GetChangedRoot();
+            var newDocument = document.WithSyntaxRoot(newRoot);
+
+            return newDocument;
+
+            //Creates a new parameter symbol for all variables captured in the local function
+            static ImmutableArray<IParameterSymbol> CreateParameterSymbol(ImmutableArray<ISymbol> captures)
+            {
+                var parameters = ArrayBuilder<IParameterSymbol>.GetInstance(captures.Length);
+
+                foreach (var symbol in captures)
+                {
+                    var type = symbol.GetSymbolType();
+                    parameters.Add(CodeGenerationSymbolFactory.CreateParameterSymbol(
+                        attributes: default,
+                        refKind: RefKind.None,
+                        isParams: false,
+                        type: type,
+                        name: symbol.Name.ToCamelCase().TrimStart(s_underscore)));
```

Comment:
We probably don't need to trim the underscore here (it was for fields), locals usually don't have this prefix.